### ISSUE TITLE
CFIXX Code Error

### ISSUE THREAD

**Issue by mustakimur** [HexHive/CFIXX#4]
I have found following code error in CodeGenFunction::EmitCXXDeleteExpr(const CXXDeleteExpr *E) in /clang/lib/CodeGen/CGExorCXX.cpp file:
1.
`llvm::Value *thisPtr = Builder.CreateBitCast(This.getPointer(), CGM.VoidPtrTy);thisPtr = Builder.CreatePtrToInt(This.getPointer(), CGM.Int64Ty);`
_This_ is not defined locally or globally, should it be _Ptr_?
2.
`if (VTable->getType() != VTableTy)VTable = dyn_cast<llvm::Instruction>(Builder.CreateBitCast(VTable, VTableTy));`
_VTableTy_ is not defined locally or globally.

**Comment by nburow2** [comment]
Using the scripts in the repo, ie ./configure && make the CFIXX compiler built without error.  As mentioned on issue 3, I'm still debugging building libcxx.

**Comment by mustakimur** [comment]
You mean code in CGExprCXX.cpp is fine?

**Comment by nburow2** [comment]
Well - in so much as it builds yes.  I did not rerun spec etc.

This is all closely tied to llvm-3.9.1 (the latest release a year ago when I was actively developing this).  The scripts take care of this though, pulling down the correct version and then applying my changes.  I realize this is a somewhat unusual structure, but we were optimizing for smaller repos where it was easy to find what files had been changed.
